pandas/core/window/aggregators.py

```diff
@@ -160,32 +162,83 @@ def combine(self, other) -> None:
         self.total += other.total
         self.count += other.count
 
-    @classmethod
-    def make_aggregator(cls, values: np.ndarray, min_periods: int) -> BaseAggregator:
-        aggregator = SubtractableAggregator(values, min_periods, cls())
+    def make_aggregator(self, values: np.ndarray, min_periods: int) -> BaseAggregator:
+        aggregator = SubtractableAggregator(values, min_periods, self)
         return aggregator
 
 
+sum_spec = (("count", numba.uint64), ("total", numba.float64))
+
+
+@numba.jitclass(sum_spec)
 class Mean(Sum):
     def finalize(self) -> Optional[float]:  # type: ignore
         if not self.count:
             return None
         return self.total / self.count
 
 
-def rolling_aggregation(
+agg_type.define(Mean.class_type.instance_type)
+
+aggregation_signature = (numba.float64[:], numba.int64[:], numba.int64[:], numba.int64)
+
+
+@numba.njit(aggregation_signature, nogil=True, parallel=True)
+def rolling_mean_class(
     values: np.ndarray,
     begin: np.ndarray,
     end: np.ndarray,
     minimum_periods: int,
-    kernel_class,
+    # kernel_class,  Don't think I can define this in the signature in nopython mode
 ) -> np.ndarray:
     """Perform a generic rolling aggregation"""
-    aggregator = kernel_class.make_aggregator(values, minimum_periods)
+    aggregator = Mean().make_aggregator(values, minimum_periods)
+    # aggregator = kernel_class().make_aggregator(values, minimum_periods)
     result = np.empty(len(begin))
     for i, (start, stop) in enumerate(zip(begin, end)):
         result[i] = aggregator.query(start, stop)
     return result
 
 
-rolling_mean = partial(rolling_aggregation, kernel_class=Mean)
+@numba.njit(aggregation_signature, nogil=True, parallel=True)
+def rolling_mean_method(
```

Comment:
Need to move this into a separate file.

1) One file should have the jitclass implementations (kernels and aggregators)
2) One file will have the njit implementations (current cython "style")